packages/services/cdn-worker/src/artifact-storage-reader.ts

```diff
@@ -20,8 +20,27 @@ type SDLArtifactTypes = `sdl${'.graphql' | '.graphqls' | ''}`;
 
 export type ArtifactsType = SDLArtifactTypes | 'metadata' | 'services' | 'supergraph';
 
+/** S3 key for stored operation body (used by CDN). */
+export function buildOperationS3BucketKey(
+  targetId: string,
+  appName: string,
+  appVersion: string,
+  hash: string,
+) {
+  return ['app', targetId, appName, appVersion, hash].join('/');
+}
```

Comment:
Maybe we should assert all values are available here? To avoid a situation where we publish to `app/undefined` or `app` or any shared space.